src/main/java/Client.java

```diff
@@ -161,21 +161,51 @@ void changeRemoteWorkingDir() throws SftpException {
   }
 
   /**
-  * Upload file to current remote directory path
-  */
+   * Upload file to current remote directory path
+   */
   void uploadFile(String filename) throws SftpException {
     cSftp.put(filename, filename);
     String pwd = cSftp.pwd();
     out.println("The file has been uploaded to: " + pwd);
   }
 
   /**
-  * Download file to current local directory path
-  */
+   * Download file to current local directory path
+   */
   void downloadFile(String filename) throws SftpException{
     cSftp.get(filename,filename);
     String lpwd = cSftp.lpwd();
     out.println("The file has been downloaded to: " + lpwd);
   }
 
+  /**
+   * Rename file on remote directory
+   */
+  void renameRemoteFile() throws SftpException {
+    out.println("Enter the original file name: ");
+    String filename = scanner.next();
+    out.println("Enter the new file name: ");
+    String newFilename = scanner.next();
+    cSftp.rename(filename, newFilename);
+    out.println(filename + " has been renamed to: " + newFilename + "\n");
+  }
+
+  /**
+   * Rename file on local directory
+   */
+  void renameLocalFile() {
+    out.println("Enter the original file name: ");
+    String filename = scanner.next();
+    out.println("Enter the new file name: ");
+    String newFilename = scanner.next();
+
+    File file = new File(filename);
+    File newFile = new File(newFilename);
+
+    if (file.renameTo(newFile)) {
```

Comment:
I am assuming this method also deletes the dupe since it looks like there are now two files (file and newFile) with the same name.